Auth/src/commonMain/kotlin/io/github/jan/supabase/auth/status/NotAuthenticatedReason.kt

```diff
@@ -0,0 +1,35 @@
+package io.github.jan.supabase.auth.status
+
+import io.github.jan.supabase.auth.exception.AuthErrorCode
+
+/**
+ * Represents the reason for an [SessionStatus.NotAuthenticated] status.
+ */
+sealed interface NotAuthenticatedReason {
```

Comment:
Thinking about removing this entirely and just using the two parameters in the `NotAuthenticated` status: `isSignOut`, `error`